index.js

```diff
@@ -1,7 +1,30 @@
 'use strict';
 
+function defaultArgv() {
+  // When evaluating script arg, user CLI args follow executable
+  const execArgv = process.execArgv;
+  if (execArgv.includes('-e') || execArgv.includes('--eval') ||
+      execArgv.includes('-p') || execArgv.includes('--print')) {
+    return process.argv.slice(1);
+  }
+
+  // In bundled Electron app, user CLI args follow executable:
+  // 1) process.versions.electron is either set by electron, or undefined
+  //    see https://github.com/electron/electron/blob/master/docs/api/process.md#processversionselectron-readonly
+  // 2) process.defaultApp is either set by electron in an electron
+  //    unbundled app, or undefined
+  //    see https://github.com/electron/electron/blob/master/docs/api/process.md#processversionselectron-readonly
+  if (process.versions && process.versions.electron && !process.defaultApp) {
```

Comment:
I could imagine getting some push back on this when we merge into Node.js, because it's so specific to one technology bundling Node.js.

electron is so popular I would argue on the behalf of this feature though.

Reply:
Good comment about push back. Similarly, I was hesitant about adding Electron support to Commander to work around an historical Electron mistake.

I mentioned `process.mainArgs` in the PR. Actually, `process.mainArgs` could cover Electron too, since Electron can implement it to be aware of the Electron launch scenarios.

I'll refactor `defaultArgv` to be a placeholder for `process.mainArgs ` to make it clearer that the functionality is wanted but may need to be part of `parseArgs`.